pkg/session/bootstrap.go

```diff
@@ -1369,6 +1372,77 @@ var (
 	SupportUpgradeHTTPOpVer int64 = version174
 )
 
+func acquireLock(s sessiontypes.Session) bool {
+	dom := domain.GetDomain(s)
+	if dom == nil {
+		logutil.BgLogger().Warn("domain is nil")
+		return false
+	}
+	cli := dom.GetEtcdClient()
+	if cli == nil {
+		logutil.BgLogger().Warn("etcd client is nil")
+		return false
+	}
+	// The lock is used to make sure only one TiDB server is bootstrapping the system.
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	lease, err := cli.Grant(ctx, 30)
+	cancel()
+	if err != nil {
+		return false
+	}
+	etcdSession, err := concurrency.NewSession(cli, concurrency.WithLease(lease.ID))
+	if err != nil {
+		return false
+	}
+	mu := concurrency.NewMutex(etcdSession, bootstrapOwnerKey)
+	err = mu.Lock(context.Background())
+	return err == nil
+}
+
+func releaseLock(s sessiontypes.Session) {
+	dom := domain.GetDomain(s)
+	if dom == nil {
+		logutil.BgLogger().Warn("domain is nil")
+		return
+	}
+	cli := dom.GetEtcdClient()
+	if cli == nil {
+		logutil.BgLogger().Warn("etcd client is nil")
+		return
+	}
+	etcdSession, err := concurrency.NewSession(cli)
+	if err != nil {
+		return
+	}
+	mu := concurrency.NewMutex(etcdSession, bootstrapOwnerKey)
+	err = mu.Unlock(context.Background())
+	if err != nil {
```

Comment:
What if release lock meet error? Do we have method to make sure the lock always will be released successfully?